vllm/v1/worker/gpu_worker.py

```diff
@@ -329,25 +327,12 @@ def execute_model(
             assert isinstance(output, IntermediateTensors)
             get_pp_group().send_tensor_dict(output.tensors,
                                             all_gather_group=get_tp_group())
-            output = EMPTY_MODEL_RUNNER_OUTPUT
+            empty_output = copy.copy(EMPTY_MODEL_RUNNER_OUTPUT)
```

Comment:
output here will include the finished_sending and recving, so you basically need to pass it through to the empty output in this if condition for PP

Reply:
Don't need to do copying in non-connector case
```python
    empty_output = EMPTY_MODEL_RUNNER_OUTPUT
    if output.finished_sending or output.finished_recving:
        empty_output = copy.copy(empty_output)
        empty_output.finished_sending = output.finished_sending
        empty_output.finished_recving = output.finished_recving
    output = empty_output        
```

Reply:
My understanding of PP case was not 100% correct. I think now things will work. Extended IntermediateTensors to include finished_sending and finished_recving so that we can pass those through to construct the empty output here.